Reworked dbpedia scripts

TimHenkelmann opened the pull request:
Had to change the following things:
- added "done ...." to process.send message such that it can be checked in cli.js
- add checks for "body" return value of request, otherwise cheerio throws error
- fixed dateOfBirth and dateOfDeath ( needs to be $(this) in the each loop)
- auto-formating

TimHenkelmann committed:
reworked dbpedia scripts

kordianbruck commented on dbpedia/dbPedia_Composers.js:
```diff
@@ -20,10 +21,36 @@ function replaceURLAndUnderscore(str) {
+});
+
+
+function get_URL(value, callback) {
```
So WTF are you doing? CamelCase? UnderScores? :dancing_men:

TimHenkelmann replied:
not my code but will incorporate the changes

kordianbruck commented on dbpedia/dbPedia_Composers.js:
```diff
@@ -82,7 +109,7 @@ function getResults(queryURL, callback) {
             var pseudonym = [];
             if ($('span[property="dbp:psuedonym"]').text() && $('span[property="dbp:psuedonym"]').text() != "") {
-                $('span[property="dbp:psuedonym"]').each(function(index) {
+                $('span[property="dbp:psuedonym"]').each(function (index) {
```
When using a complex selector it pays off to save the object to a local var

kordianbruck commented on dbpedia/dbPedia_Composers.js:
```diff
@@ -123,14 +150,14 @@ function getResults(queryURL, callback) {
                 });
             }
 
             if ($('a[rel="dbo:artist"]').attr('href') && $('a[rel="dbo:artist"]').attr('href') != "") {
```
Especially if you use the same var like 20 times

TimHenkelmann replied:
Every selector is used **at most** 3 times, but you're still right. I will adapt all dbpedia scripts.

kordianbruck commented on dbpedia/dbpedia_Classical_musicians_by_century.js:
```diff
@@ -31,180 +30,183 @@ function replaceURLAndUnderscore(str) {
 }
 
 function fetch1(link1) {
```
So what does fetch1 do? Naming this function a bit better would give other people an insight into what is supposed to happen here

TimHenkelmann replied:
not my code but will incorporate the changes

kordianbruck reviewed (commented):
I stopped looking at some point as this PR is quite big and seems to have a lot of code duplication

TimHenkelmann committed:
added helper functions for dbpedia scripts

TimHenkelmann committed:
- moved scraping of dbpedia data into helper function
- introduced local variables such that selectors are only computed once
- renamed functions (fetch1, fetch2, ...) to meaningful names

TimHenkelmann committed:
remove duplicates in dbpedia scripts with multiple URLs

TimHenkelmann committed:
added "meaningful" error logging to all requests

TimHenkelmann committed:
fixed bug in "_by_century.js" which scraped the same URLs multiple times
fixed output object
fixed "unmeaningful" declarations "value2"
added split to "_by_nationality.js" name

TimHenkelmann commented:
@ShilpaGhanashyamGore @angelinrashmi2000 @LukasNavickas Done with my fixes, can you review this please?

kordianbruck reviewed (approved):
Nice rework! Excellent work <3
